src/api/foreground/foreground-setup-wallet-sdk.ts

```diff
@@ -12,102 +12,134 @@ export function setupWalletSDK(targetWindow: Window = window) {
 
-        // TODO: Declare outside (factory) to facilitate testing?
-        async function callback(e: MessageEvent<ApiResponse>) {
-          // TODO: Make sure the response comes from targetWindow.
-          // See https://stackoverflow.com/questions/16266474/javascript-listen-for-postmessage-events-from-specific-iframe.
+      // wait for result from background
+      window.addEventListener("message", callback);
 
-          let { data: res } = e;
+      // TODO: Declare outside (factory) to facilitate testing?
+      async function callback(e: MessageEvent<ApiResponse>) {
+        // TODO: Make sure the response comes from targetWindow.
+        // See https://stackoverflow.com/questions/16266474/javascript-listen-for-postmessage-events-from-specific-iframe.
 
-          // validate return message
-          if (`${data.type}_result` !== res.type) return;
+        let { data: res } = e;
 
-          // only resolve when the result matching our callID is deleivered
-          if (data.callID !== res.callID) return;
+        // validate return message
+        if (`${data.type}_result` !== res.type) return;
 
-          window.removeEventListener("message", callback);
+        // only resolve when the result matching our callID is deleivered
+        if (data.callID !== res.callID) return;
 
-          // check for errors
-          if (res.error) {
-            return reject(res.data);
-          }
+        window.removeEventListener("message", callback);
 
-          // call the finalizer function if it exists
-          if (mod.finalizer) {
-            const finalizerResult = await mod.finalizer(
-              res.data,
-              foregroundResult,
-              params
-            );
-
-            // if the finalizer transforms data
-            // update the result
-            if (finalizerResult) {
-              res.data = finalizerResult;
-            }
-          }
+        // check for errors
+        if (res.error) {
+          return reject(res.data);
+        }
 
-          // check for errors after the finalizer
-          if (res.error) {
-            return reject(res.data);
+        // call the finalizer function if it exists
+        if (mod && mod.finalizer) {
+          const finalizerResult = await mod.finalizer(
+            res.data,
+            foregroundResult,
+            params
+          );
+
+          // if the finalizer transforms data
+          // update the result
+          if (finalizerResult) {
+            res.data = finalizerResult;
           }
+        }
 
-          // resolve promise
-          return resolve(res.data);
+        // check for errors after the finalizer
+        if (res.error) {
+          return reject(res.data);
         }
-      });
+
+        // resolve promise
+        return resolve(res.data);
+      }
+    });
   }
 
+  /**
+   * Create the Proxy object.
+   */
+  const proxyWallet = new Proxy<Record<string, any>>(
+    {},
```

Comment:
We probably want to prepopulate this:

```
for (const mod of foregroundModules) {
  walletAPI[mod.functionName] = (...params: any[]) =>{
    return callForegroundThenBackground(mod.functionName, ...params);
  }
}
```

Then:

```
const proxyWallet = new Proxy<Record<string, any>>(walletAPI, { ... });
```

This way also the `Reflect.get(target, propKey)` will actually be used.